File: src/Software Engineer Prepare Kit/14/solution.ts
export default function debounceTimestamps(timestamps: number[], K: number): number {
	if (!timestamps.length) return 0

	let lastSaved = 0
	let read = 1

	while (read < timestamps.length) {
		if (timestamps[read]! - timestamps[lastSaved]! >= K) {
			timestamps[lastSaved + 1] = timestamps[read]!
			lastSaved++
		}

		read++
	}

	return lastSaved + 1
}

File: src/Software Engineer Prepare Kit/14/solution.go
package main

func DebounceTimestamps(timestamps []int32, K int32) int32 {
	if len(timestamps) == 0 {
		return 0
	}

	var read int32 = 1
	var lastSaved int32 = 0

	for read < int32(len(timestamps)) {
		if timestamps[read]-timestamps[lastSaved] >= K {
			timestamps[lastSaved+1] = timestamps[read]
			lastSaved++
		}

		read++
	}

	return lastSaved + 1
}
